doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,246 @@
+
+#### Story 4
+
+Scorecard would process tests based on an on-disk directory or bundle image that contains the scorecard configuration.
+
+#### Story 5 
+
+- I am as an Operator developer, would like to be able to create tests using the same Assert syntax adopted by [Kuttl](https://github.com/kudobuilder/kuttl) and check its results aggregated within scorecard results. 
+
+**Implementation Notes**
+
+The Scorecard would support the creation of a Kuttl test image and the execute tests via Kuttl, however, it would be transparent for users since its results will be integrated into scorecard results.
+
+## Design Details
+
+### Glossary
+
+ * test image - this is a container image built by end-users to contain a custom test implementation
+ * test configuration (DSL) - this is a file that contains the test definitions and labels which determine what tests are run and how they are configured
+ * Custom Test Definition - A custom test is any sort of test logic required, written in any programming language that satifies the scorecard test requirements for output format and execution.
+
+#### Test Image
+
+A custom test is based on a container image, that image is created by
+the end-user using whatever tools they want to use.
+
+The test image must produce container log output that conforms to the scorecard Test Result output.  The output is an array of Test Results, so a test image could contain multiple tests if necessary.  Here is a sample of test output in the required format:
+```
+    [
+      {
+	      "name": "Spec fields with descriptors",
+	      "description": "All spec fields have matching descriptors in the CSV",
+	      "state": "fail",
+	      "suggestions": [
+		"Add a spec descriptor for size"
+	      ],
+	      "crname": "example-memcached"
+      }, 
+      {
+	      "name": "Some other test",
+	      "description": "Another test on the CSV",
+	      "state": "pass",
+	      "suggestions": [
+	      ],
+	      "crname": "example-memcached"
+      },
+    ]
+```
+
+The test image will be executed as a Pod by the scorecard with a restart policy of `never`.
+
+Also, for each test execution, scorecard would also make the
+bundle image contents available to test images by mounting
+the bundle image contents into the following location inside the test pod:
+
+```
+/scorecard
+```
+
+This allows tests to have access to all the bundle contents if they 
+are required for the test logic.  Note:  It is the responsibility of the test to iterate though each CR that they want to test and create Custom Resources as scorecard will no longer be responsible for this.  The list of CRs to test are found in the CSV within the bundle image content.
+
+For the case of executing a test, the bundle image will be passed as a scorecard command flag:
+```
+operator-sdk scorecard --bundle='quay.io/myorg/mytest' --selector=suite=basic
+```
```

Comment:
What was the decision about `--bundle` vs `--test-assets`, or maybe even just `--mount`? Basically a flag like `--bundle` but not specific to OLM.

I thought we arrived at something more generic so as to not use/require OLM terminology.

We _would_ enable the OLM test configuration if and only if we find the expected OLM bundle directories in the expected location within the mounted docker image/directory.

And we _would_ enable the kuttl test configuration if and only if we find the expected kuttle test directories in the expected location within the mounted docker image/directory.

Reply:
I was thinking we would have a single flag that would cover both cases a) on-disk directory and b) bundle image, scorecard would look at what is passed to know if the contents are on-disk (exploded) or in an image that needs to be extracted to on-disk.  I'm totally flexible on the name, maybe --test-assets as Joe suggested?  I could also make it 2 flags is that is a better end user experience.

Reply:
Part of the plan is to drastically simplify config/flag sprawl from the current scorecard. My vote is that if we want to keep `--bundle` lets just use that for now (and it can accept a bundle image or bundle directory).

Then if we want to become more generic later, we can add a new flag and optionally deprecate `--bundle` if the new flag incorporates the features of `--bundle`. WDYT?